zapcore/sampler.go

```diff
@@ -126,9 +183,47 @@ func (s *sampler) Check(ent Entry, ce *CheckedEntry) *CheckedEntry {
 	}
 
 	counter := s.counts.get(ent.Level, ent.Message)
-	n := counter.IncCheckReset(ent.Time, s.tick)
+	n, prevN := counter.IncCheckReset(ent.Time, s.tick)
 	if n > s.first && (n-s.first)%s.thereafter != 0 {
 		return ce
 	}
+	if s.reporting.enabled {
+		err := s.reporting.report(s.reporting.core, ent, s.getSampleCount(prevN))
+		if err != nil {
+			fmt.Printf("%v sampling report write error: %v\n", time.Now(), err)
```

Comment:
This is still writing directly to stdout, you need to at least get a hold of `zap.Logger.errorOutput` and write to it instead (especially if we cannot propagate the error).

nit: always prefer tighter `if err := f(); err != nil {}` scoping when possible.